File: com/pdfjet/TextAnnotation.java
/**
 * TextAnnotation.java
 *
 * Copyright (c) 2025 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
package com.pdfjet;

public class TextAnnotation extends BaseAnnotation {
    public TextAnnotation() {
        super.annotationType = Annotation.Text;
    }

    public void Rotate(double degrees) {
        float[] center = container.getRotationCenter();
        super.point1 = Container.rotateAroundCenter(super.point1, center, degrees);
        super.point2 = Container.rotateAroundCenter(super.point2, center, degrees);
    }
}

File: net/pdfjet/TextAnnotation.cs
/**
 * TextAnnotation.cs
 *
 * Copyright (c) 2025 PDFjet Software
 * Licensed under the MIT License. See LICENSE file in the project root.
 */
using System;

namespace PDFjet.NET {
public class TextAnnotation : BaseAnnotation {
    public TextAnnotation() {
        base.annotationType = Annotation.Text;
    }

//    public void Rotate(double degrees) {
//        float[] center = container.GetRotationCenter();
//        base.point1 = Container.RotateAroundCenter(base.point1, center, degrees);
//        base.point2 = Container.RotateAroundCenter(base.point2, center, degrees);
//    }
}
}
